velox/functions/lib/string/StringCore.h

```diff
@@ -261,7 +261,7 @@ static int64_t findNthInstanceByteIndexFromStart(
 
   // Find next occurrence
   return findNthInstanceByteIndexFromStart(
-      string, subString, instance - 1, byteIndex + subString.size());
+      string, subString, instance - 1, byteIndex + 1);
```

Comment:
What issue is being fixed here?

Reply:
Fix the overlapped issue.   The result of `substring_index("||||||", "|||", 3)` will be` || `with this PR and `||||||` without this PR.

Reply:
Is this an existing bug? Which functions are affected? It would be nice to extract this fix into a separate PR.

Reply:
Filed https://github.com/facebookincubator/velox/pull/6410.

Reply:
@mbasmanova  https://github.com/facebookincubator/velox/pull/6410 merged.